gobblin-utility/src/test/java/org/apache/gobblin/util/AvroUtilsTest.java

```diff
@@ -279,4 +280,14 @@ public void testComplexTypesInUnionNotSupported()
 
     AvroUtils.getFieldSchema(record.getSchema(), TEST_LOCATION);
   }
+
+  @Test
+  public void testGetAvroFileSampleInDirectoryHelper() throws Exception {
+    // This resource path, on purpose, has set b.avro to be the newest avro file.
+    Path avroDir = new Path(this.getClass().getClassLoader().getResource("nestedAvroDir").toURI());
+    FileStatus fileStatus = AvroUtils.getAvroFileSampleInDirectory(avroDir, FileSystem.get(new Configuration()), true);
+    Assert.assertTrue(fileStatus.getPath().getName().equals("b.avro"));
```

Comment:
This test is relying on file modification times, correct? I don't think they will be preserved by git, so this test could fail in the future. To make this test robust, you should programmatically set the modification time of the "latest" file in your test before calling getAvroFileSampleInDirectory.